src/fromager/sources.py

```diff
@@ -281,6 +284,18 @@ def patch_source(ctx: context.WorkContext, source_root_dir: pathlib.Path) -> Non
             )
 
 
+def _warn_for_old_patch(source_root_dir: pathlib.Path, version: Version) -> None:
+    # Get the existing version and req name as per the source_root_dir naming conventions
+    req_name = source_root_dir.name.rsplit("-", 1)[0]
+    existing_version = Version(source_root_dir.name.rsplit("-", 1)[1])
+
+    # Compare the version object to see if patches for older version exists
+    if version > existing_version:
```

Comment:
We only want to emit these warnings if old patches exist and patches for the current version do not exist, because that might mean we need to update those patches to work with the newer version of the package.